File: MainApp/ZuulAndFrontend/Frontend/src/app/model/message.ts
import { UserInfo } from "./userInfo";
import { User } from "./user";
import { RentRequest } from "./rentRequest";

export class Message {
  id: number;
  text: string;
  sender: UserInfo;
  dateAndTime: string;
  rentRequest: RentRequest;

  constructor(
    text: string,
    sender: UserInfo,
    dateAndTime: string,
    rentRequest: RentRequest,
    id?: number
  ) {
    this.text = text;
    this.sender = sender;
    this.rentRequest = rentRequest;
    this.dateAndTime = dateAndTime;
    this.id = id;
  }
}

File: MainApp/Renting/src/main/java/jvn/Renting/model/Message.java
package jvn.Renting.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String text;

    @Column(nullable = false)
    private Long senderId;

    @Column(nullable = false)
    private String senderEmail;

    @Column(nullable = false)
    private LocalDateTime dateAndTime;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private RentRequest rentRequest;
    
}
